src/routers/account.js

```diff
@@ -616,16 +618,48 @@ router.post('/checkout/:priceId', async (request, response) => {
 
 	await database.PNID.updateOne({ pid }, {
 		$set: {
-			connections: {
-				stripe: {
-					customer_id: customer.id // ensure PNID always has latest customer ID
-				}
-			}
+			'connections.stripe.customer_id': customer.id // ensure PNID always has latest customer ID
 		}
 	}, { upsert: true }).exec();
 
 	const priceId = request.params.priceId;
 
+	const pnid = await database.PNID.findOne({ pid });
+
+	if (pnid.get('access_level') >= 2) {
+		response.cookie('error', 'Staff members do not need to purchase tiers', { domain: '.pretendo.network' });
+		return response.redirect('/account');
+	}
+
+	const stripeConnection = pnid.get('connections.stripe');
+
+	if (stripeConnection.subscription_id) {
+		// User already has a subscription with us
+		if (stripeConnection.price_id === priceId) {
+			response.cookie('error', 'Already subscribed to this tier', { domain: '.pretendo.network' });
+			return response.redirect('/account');
+		}
```

Comment:
this information can be outdated at the time of execution. (webhooks may not have fired yet for example)
use the fetched subscription to do this check.